clap_complete/src/dynamic/completer.rs

```diff
@@ -263,6 +263,86 @@ fn complete_arg(
                             .visible(true)
                         }),
                 );
+            } else if let Some(mut short) = arg.to_short() {
+                if !short.is_negative_number() {
```

Comment:
I think at this point we should be breaking some of these branches out into functions.

For this specific case
- It would help us with doing an early return with `is_negative_number`, rather than shifting things to the right
- We could look at reusing code for short processing with `complete`

Reply:
Thanks for your review! I will refactor the code to code reuse.